src/pdmat.jl

```diff
@@ -1,19 +1,19 @@
 """
 Full positive definite matrix together with a Cholesky factorization object.
 """
-struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
+struct PDMat{T<:Real,S<:AbstractMatrix,C<:Factorization} <: AbstractPDMat{T}
     mat::S
-    chol::Cholesky{T,S}
+    chol::C
```

Comment:
Not sure if this will be a breaking change, but if so should we change the fieldname to `fmat` or `fact` or something?

Reply:
This PR will be a breaking change anyway, so we could also rename the field. I guess I would prefer `fact`.